File: app/src/main/java/com/markusw/cosasdeunicorapp/data/model/Message.java
package com.markusw.cosasdeunicorapp.data.model;

public class Message {

    private String content;
    private User sender;
    private long timestamp;

    public Message() {
    }

    public Message(String content, User sender, long timestamp) {
        this.content = content;
        this.sender = sender;
        this.timestamp = timestamp;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public User getSender() {
        return sender;
    }

    public void setSender(User sender) {
        this.sender = sender;
    }

    public long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(long timestamp) {
        this.timestamp = timestamp;
    }
}

File: app/src/main/java/com/markusw/cosasdeunicorapp/home/domain/model/Message.kt
package com.markusw.cosasdeunicorapp.home.domain.model

import com.markusw.cosasdeunicorapp.core.domain.model.User

data class Message(
    val content: MessageContent = MessageContent(text = "Example message"),
    val sender: User = User(),
    val timestamp: Long = 0L
)

data class MessageContent(
    val text: String = "",
    val type: MessageType = MessageType.PlainText,
    val replyTo: Message? = null,
    val imageUrl: String? = null
)